libs/api/transaction/data-access/src/lib/api-transaction-data-access.service.ts

```diff
@@ -156,6 +156,10 @@ export class ApiTransactionDataAccessService implements OnModuleInit {
       throw new UnauthorizedException('Request not allowed')
     }
 
+    if (appEnv?.ipsAllowed.length > 0 && !appEnv?.ipsAllowed.includes(ip)) {
+      throw new UnauthorizedException('Request not allowed')
+    }
+
```

Comment:
I would like to confirm which filter take priority here, but the logic I applied is this:

1. If the IP is in the blocked list, we block it. 
2. If it is not there, we check if we have an allowed list
3. If it is not in the allowed list, we block it.

Reply:
My idea was to make the `allow` list leading:

If the `allowed` list exists, it will allow if your IP is in that list, any other IP is blocked. This is useful if you want to create an environment that you only use on your server, or if you want to test something and only allow your home IP.

If the `blocked` list exists, it will block if your IP is in that list, any other IP is allowed. This is useful for blocking bad actors.

If none of the lists exist, it will allow your request.

In the future, we want to look at allowing and blocking certain ranges, but that's not for now.